### DISCUSSION TITLE
more c-syntax/operators

### DISCUSSION THREAD

**Discussion by vbBaka** [twinbasic/twinbasic#612] [Ideas]
I notice most people enjoy c,
so we need more c-operators/functions to have alternatives to all those abominations like mod, str, len, cos, sin, tan, asc, abs, chr, ltrim, rtrim, well theres a lot.
people don't like abbreviations, since its not "readable enough". they enjoy symbols. 
we should start adding alternatives, so more c people will join the project.

modulo % is one. and Im sure the c people in here can give more examples for what we can add.
unfortunately TB want full back compatibility, so theres a problem here, it will pollute if we have 2 alternatives.
but I think its quite inconsistent to mix basic and c like we have. 
but since most people enjoy c, we should weight for their benefit, basic people need to adapt.

**Comment by WaynePhillipsEA** [comment]
I appreciate your frustrations with the earlier discussion regarding the shift operators.  However, I'd advise taking a step back and keep things sensible.  Nobody has suggested that we are targeting existing C developers with twinBASIC.   We are not.  We are simply trying to do what we think is best to advance the language.  

We don't have the budget of Microsoft's R&D department, and so we have to make do with what we have available to us... the community here.   By having language/syntax discussions openly in the community we hope to make the right decisions that reflect the majority.   There's never going to be full agreement on those types of things, but it's important that we respect the decisions of the community, even if we don't personally agree with them.

I would hope that we can have a mature, friendly, and lively debate without resulting in threads like this.

**Comment by wqweto** [comment]
Btw, it's funny the discussion starts with `Mod` vs `%` because in C/C++ the modulo operator returns the "remainder of the division" not the math modulo.

This explains why `-123 % 10` in C/C++ equals -3 and not 7 as the original math modulo operation is never negative.

If you check VBx `Mod` operator with the negative expression above you will find if it's modulo or remainder as it's currently implemented. 

Btw, in python for instance `%` is real modulo and returns 7.
